src/tiktok_uploader/upload.py

```diff
@@ -351,14 +353,16 @@ def _set_video(driver, path: str = '', num_retries: int = 3, **kwargs) -> None:
                 )
```

Comment:
For unknown reason, sometimes Selenium will not find the `//video` element even though it is visible on the page and it will result with an `Timeout Exception` I don't why this happen but I decided to change this line on my fork with :
```
(By.XPATH, "//button[contains(text(), 'Change video')]")
```
 which also detect that the video has been fully uploaded. I have also added a `time.sleep(1)` after the WebDriverWait just in case.

My method seems to work but might not be consistent with the your code, that is why I have not pushed any changes about that.